frontend/src/Pages/CreateProgram.tsx

```diff
@@ -0,0 +1,400 @@
+import { useNavigate } from 'react-router-dom';
+import {
+    CreditType,
+    Facility,
+    ProgramStatus,
+    ProgramType,
+    ToastState,
+    FundingType
+} from '@/common';
+import { SubmitHandler, useForm, Controller } from 'react-hook-form';
+import { TextInput, TextAreaInput } from '@/Components/inputs';
+import API from '@/api/api';
+import Select, {
+    StylesConfig,
+    CSSObjectWithLabel,
+    GroupBase
+} from 'react-select';
+import { canSwitchFacility, useAuth } from '@/useAuth';
+import { useLoaderData } from 'react-router-dom';
+import { useToast } from '@/Context/ToastCtx';
+
+interface ProgramInputs {
+    name: string;
+    description: string;
+    credit_type: GenericOption<CreditType>[];
+    program_type: GenericOption<ProgramType>[];
+    program_status: ProgramStatus;
+    funding_type: GenericOption<FundingType>;
+    facilities: GenericOption<number>[];
+}
+
+interface TransformedProgramInput {
+    name: string;
+    description: string;
+    credit_type: CreditType[];
+    program_type: ProgramType[];
+    program_status: ProgramStatus;
+    funding_type: FundingType;
+    facilities: number[];
+}
+
+interface ProgramResponse {
+    message: string;
+    program_id: number;
+}
```

Comment:
See my comment in `programs_handler`. We really only need this because we are returning that custom response, right? Can we return the whole program instead?